### PR TITLE
fix cell indexing

### PR DESCRIPTION
**Pull request by rubengrandia**
trick is to move the half shift (width / 2) into the rounding operation to work for both even and odd numbered maps. 

We also need to correct for the fact that the center of a cell is shifted by half the resolution when viewed from the center. That is where you get the (width - 1.0) / 2 from

### CODE REVIEW

**Comment by rubengrandia** [comment]
Old code:
Screenshot from 2022-05-04 22-37-50

New code:
Screenshot from 2022-05-04 22-35-54

**Comment by lorenwel** [comment]
Isn't the `round` the problem here? I think flooring with just `int i = int((x - center) / resolution);` should also be correct, no?

**Comment by rubengrandia** [comment]
But you are right that it should be possible to exploit the flooring of the int, instead of doing round manually. I think this one should work:
return static_cast<int>((x - center) / resolution + static_cast<float>(width) / 2);

Basically removing the half shift again and then taking the floor instead of the round. I do think it is necessary to not divide width as an integer to be correct for both even and odd sized maps.

**Comment by lorenwel** [comment]
I meant more the problem with the original implementation. 
[This is what I meant](http://cpp.sh/4jvf3).
So 
```
int get_x_idx_floor(float x, float center, float resolution, int width) {
    return static_cast<int>((x - center) / resolution + width*0.5);
}
```

**Comment by lorenwel** [comment]
Ok, we came to the same conclusion :P 
I don't have a strong preference which one to use, I could imagine that integer casting is slightly faster. 
In any case, I would replace that `/ 2` with a `* 0.5` :D

**Review by lorenwel — APPROVED** [review]
If you tested this, I'm fine merging.

**Comment by rubengrandia** [comment]
I tested on the robot yesterday, looks fine. Let's wait for @mktk1117 's opinion. He can also run it on some larger offline dataset to double check.

**Comment by mktk1117** [comment]
Tested from my side and it looks good.

**Review by mktk1117 — APPROVED** [review]

File: elevation_mapping_cupy/script/custom_kernels.py [hunk 1]
```diff
@@ -16,9 +16,16 @@ def map_utils(resolution, width, height, sensor_noise_factor, min_valid_distance
         __device__ float16 round(float16 x) {
             return (int)x + (int)(2 * (x - (int)x));
         }
-        __device__ int get_xy_idx(float16 x, float16 center) {
+        __device__ int get_x_idx(float16 x, float16 center) {
             const float resolution = ${resolution};
-            int i = round((x - center) / resolution);
+            const float width = ${width};
+            int i = round((x - center) / resolution + (width - 1.0) / 2);
```
[the remark this answers is not in the document]
Comment (rubengrandia) [reply elevation_mapping_cupy/script/custom_kernels.py]: I was not 100% sure about the syntax of these kernels. So I followed your lead in the line above.

**Pull request merged**